### DISCUSSION TITLE
How to list the unit type and respective aliases

### DISCUSSION THREAD

**Discussion by luzfcb** [josdejong/mathjs#2985] [Q&A] [answered]
Hello, what is the correct way to get an array of all MASS units that mathjs supports?
I'm developing a simple measure conversion application.

I want to show in a `select` the mass unit list for the user to select. I want to get them directly from MathJS. Is there any out-of-the-box way to do this in the MathJS API? If not, how would you do it?

**Comment by josdejong** [comment, marked as answer]
You can loop through all units defined in `math.Unit.UNITS` and filter those that have a base unit MASS, something like:

```js
Object.values(math.Unit.UNITS).filter(unit => unit.base.key === 'MASS')
```

Note that each of the units can be prefixed too, like `milligram` and `kilogram` etc.
